apps/web/src/constants/permissionConstants.ts

```diff
@@ -1,41 +1,166 @@
 import { ROLE_NAME, RoleName } from "@gc-digital-talent/auth";
 
-const permissionConstants: Readonly<Record<string, RoleName[]>> = {
-  viewCandidates: [
+const permissionConstants = () => {
+  const viewCandidates: RoleName[] = [
```

Comment:
Is there a reason this was changed back to a function? Is it not just plain old arrays? We could just return the readonly record so we don't need to invoke a function call every time we read a permission

Reply:
🤷

Reply:
I'm pretty sure this code is from when this was originally a function. That got changed in #12065 but hasn't been properly merged here. Long story short, I'd rather this go back / stay as plain old arrays.

Reply:
Aye aye, https://github.com/GCTC-NTGC/gc-digital-talent/pull/12007/commits/2626e777b3d42106e0f12c7e2c9354afffc604f7